spacy/matcher/matcher.pyx

```diff
@@ -949,24 +951,24 @@ class _SetPredicate:
             if self.regex:
                 value = self.vocab.strings[value]
                 return any(bool(v.search(value)) for v in self.value)
-            elif value in self.value:
-                return True
             elif self.fuzzy is not None:
                 value = self.vocab.strings[value]
                 return any(self.fuzzy_compare(value, self.vocab.strings[v], self.fuzzy)
                            for v in self.value)
+            elif value in self.value:
+                return True
```

Comment:
@adrianeboyd isn't checking for exact match before trying fuzzy match much cheaper?

Reply:
It would be cheaper, but it kind of violates what we're saying that `FUZZY` does because it doesn't use the `fuzzy_compare` function.

I could imagine custom `fuzzy_compare` methods that explicitly don't match on exact matches or require a minimum edit distance? In the end it's more of a "custom comparison predicate that can be used for fuzzy matching". (I kind of wonder whether we should rename it and add the equivalent of `FUZZY0`. On the other hand I think it's a lot clearer for typical users if it's called `FUZZY`.)

Reply:
ok, I hadn't considered any cases where `fuzzy_compare` would be used to block exact matches. using that in combination with `NOT_IN` and `OP: !` would get really confusing!

I'd viewed `fuzzy_compare` as a (more expensive) fallback when exact match fails.

Reply:
I'm imagining that you can use `1-9` as a "mode" rather than a distance and have all kinds of weird custom matching.